addons/point_of_sale/static/src/app/report_service.js

```diff
@@ -0,0 +1,33 @@
+/** @odoo-module */
+
+import { registry } from "@web/core/registry";
+import { downloadReport } from "@web/webclient/actions/reports/utils";
+
+export const reportService = {
+    dependencies: ["rpc", "user", "ui"],
+    start(env, { rpc, user, ui }) {
+        const reportActionsCache = {};
+        return {
+            async download(reportXmlId, active_ids) {
+                ui.block();
+                try {
+                    reportActionsCache[reportXmlId] ||= rpc("/web/action/load", {
```

Comment:
if you are sure that your action only depends on its bare name, and not, for instance, on the context

Reply:
There are only 2 report actions used in the pos currently and neither depend on the context, so I think it's fine for now. If this changes in the future we can always add more stuff to the cache key like the action service but this service is not supposed to be general purpose